src/dygraph-interaction-model.js

```diff
@@ -509,31 +536,62 @@ DygraphInteraction.moveTouch = function(event, g, context) {
   var dataHeight = context.initialRange.y[0] - context.initialRange.y[1];
   swipe.dataX = (swipe.pageX / g.plotter_.area.w) * dataWidth;
   swipe.dataY = (swipe.pageY / g.plotter_.area.h) * dataHeight;
-  var xScale, yScale;
+
+  var xExtremes = g.xAxisExtremes()
+  var yExtremes = g.yAxisExtremes()
+
+  var xScale = 1.0, yScale = 1.0;
 
   // The residual bits are usually split into scale & rotate bits, but we split
   // them into x-scale and y-scale bits.
-  if (touches.length == 1) {
-    xScale = 1.0;
-    yScale = 1.0;
-  } else if (touches.length >= 2) {
+  if (touches.length >= 2 && !context.pinchOutOfExtremes) {
     var initHalfWidth = (initialTouches[1].pageX - c_init.pageX);
-    xScale = (touches[1].pageX - c_now.pageX) / initHalfWidth;
-
     var initHalfHeight = (initialTouches[1].pageY - c_init.pageY);
-    yScale = (touches[1].pageY - c_now.pageY) / initHalfHeight;
+    var minAllowed = 5
+
+    if (Math.abs(initHalfWidth) > minAllowed) {
+        // sensitiveness dampening: smaller pinches count much less
+        var damp = 1 / (Math.abs(initHalfWidth)-minAllowed)
+        
+        var nowHalfWidth = touches[1].pageX - c_now.pageX
+        xScale = (nowHalfWidth + damp) / (initHalfWidth + damp);
+    }
+
+    if (Math.abs(initHalfHeight) > minAllowed) {
+        var damp = 1 / (Math.abs(initHalfHeight) - minAllowed)
```

Comment:
AFAICT `var` has function scope, so this is an error. I’d say declare `var damp` above with the `initHalf{Width,Height}` and reuse it in both branches.